test/rust-test-util/test-result-reporting-client/src/client.rs

```diff
@@ -43,13 +44,30 @@ impl TrcClient {
             return Err(ReportResultError::UnsupportedTestType);
         }
 
-        let body = TestOperationResultDto::new(source, result, test_type, created_at);
-        let body = serde_json::to_string(&body).map_err(ReportResultError::CreateJsonString)?;
-        let request = Request::post(self.uri.clone())
-            .header(CONTENT_TYPE, APPLICATION_JSON)
-            .body(Body::from(body.clone()))
-            .map_err(ReportResultError::ConstructRequest)?;
+        // exponential backoff for max ~3 mins
+        let mut retries: u32 = 11;
+        let mut wait = Duration::from_millis(100);
+        loop {
+            let body =
+                TestOperationResultDto::new(source.clone(), result.clone(), test_type, created_at);
+            let body = serde_json::to_string(&body).map_err(ReportResultError::CreateJsonString)?;
+            let request = Request::post(self.uri.clone())
+                .header(CONTENT_TYPE, APPLICATION_JSON)
+                .body(Body::from(body.clone()))
+                .map_err(ReportResultError::ConstructRequest)?;
 
+            match self.trc_request(request).await {
+                Err(_) if retries > 0 => {
+                    retries -= 1;
+                    time::sleep(wait).await;
+                    wait *= 2
```

Comment:
It may not be necessary for this application of exponential backoff, but consider introducing some randomization such as what's described in this backoff crate: https://docs.rs/backoff/0.3.0/backoff/.